### PR TITLE
Add Jump instance policy

### PR DESCRIPTION
**Pull request by david4096**
This PR attempts a different way of managing permissions for the jump instance as compared to https://github.com/dockstore/dockstore-deploy/pull/86 which uses tags.

The intention here is to create a new policy that references the newly create JumpInstance and assigns permissions for performing SSM and EC2 actions.

That policy is associated with the `db-admin` group, which should have been created in the console ahead of time. I looked at creating groups via CF but it seems to cause [region related issues or failures when names are reused](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-iam-group.html).

### CODE REVIEW

**Comment by david4096** [comment]
I tested this (with Charles' help) by

1. adding a `db-admin` group on dev
1. deploying the stack (with the RDS parts commented out)
1. confirming a policy was made that is attached to the `db-admin` group
1. Removing myself from the `Developers` group which provides access to SSM for everything, and confirming I can't access the instance
1. Adding myself to the `db-admin` (which has read only access to SSM and EC2 by default) group and confirming I can access the instance
1. removing myself from `db-admin` and confirming I can no longer access the instance

To complete this effort we need to change the default policy for developers to not allow SSM for all.

File: database.yml.template [hunk 1]
```diff
@@ -328,6 +329,20 @@ Resources:
             apt install -y awscli
             /usr/local/bin/cfn-init -v --stack ${AWS::StackName} --resource JumpInstance --region ${AWS::Region}
 
+  JumpInstanceAdminPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: !Sub 'jump-${AWS::StackName}-admin-policy'
+      Groups: ["db-admin"]                      # this group should have been created manually
```
Comment (coverbeck) [inline database.yml.template:336]: You could make this a parameter with a default value of db-admin just to make it slightly more flexible.

File: database.yml.template [hunk 2]
```diff
@@ -328,6 +338,20 @@ Resources:
             apt install -y awscli
             /usr/local/bin/cfn-init -v --stack ${AWS::StackName} --resource JumpInstance --region ${AWS::Region}
 
+  JumpInstanceAdminPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: !Sub 'jump-${AWS::StackName}-admin-policy'
+      Groups: [Ref: DBAdminGroup]                      # this group should have been created manually
```
Comment (chmreid) [inline database.yml.template:345]: I think this should be `!Ref`

Comment (chmreid) [reply database.yml.template:345]:
```suggestion
      Groups: [!Ref DBAdminGroup]                      # this group should have been created manually
```

Comment (david4096) [reply database.yml.template:345]: The short form and the long form of the function are not very different. Both work

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html

**Pull request merged**